cinema/views.py

```diff
@@ -56,3 +80,44 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__seats_in_row")
+                        * F("cinema_hall__rows")
+                        - Count("tickets")
+                    )
+                )
+            )
+        if self.action == "retrieve":
+            queryset = queryset.select_related()
```

Comment:
I guess you can put .select_related() in a general queryset and also add fields that we are joining  explicitly